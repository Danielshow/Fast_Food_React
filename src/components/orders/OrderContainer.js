import { connect } from 'react-redux';
import Order from './Order';
import {getFoodsFromAPI, orderFood, getUserFromToken }
  from '../../store/actions/order';

export const mapStateToProps = state => {
  return {
    foods: state.ord.foods,
    loading: state.ord.loading,
    isUser: state.ord.isUser,
    email: state.ord.email
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    getFoods: () => {dispatch(getFoodsFromAPI());},
    orderFood: (food) => {dispatch(orderFood(food));},
    getUserFromToken: () => {dispatch(getUserFromToken());}
  };
};

const OrderContainer = connect(mapStateToProps, mapDispatchToProps)(Order);

export default OrderContainer;