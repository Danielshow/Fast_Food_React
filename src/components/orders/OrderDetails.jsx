import React from 'react';
import PropTypes from 'prop-types';


/**
 * @class
 */
const OrderDetails = ({orders, total, displayModal, handleRemove }) => {
  const orderedFood = orders.map(order => {
    return (
      <div className="orderedFood" key={order.id}>
        <div className="foodQuantity">
          {order.quantity}
          &nbsp;
          {order.food}
        </div>
        <div className='totalPrice'>
          {order.tprice}
          &nbsp;
          {/* eslint-disable-next-line */}
          <span className="remove" onClick={() => {handleRemove(order.id)}}>X</span>
        </div>

        <hr className="orderHR" />
      </div>
    );
  });
  return (
    <div className="ordersPanel">
      <div className="innerPanel">
        <h3> Cart </h3>
        <div className="myOrder">
          <hr className="orderHRUpper" />
          {orderedFood}
          <div className="food">
            <br />
            <div className='foodQuantity'>Total: </div>
            <div className='totalPrice'>
              â‚¦ &nbsp;
              {total}
            </div>
            <hr className="orderHR" />
          </div>
          <button
            type="submit"
            className="foodButton"
            onClick={() => displayModal()}
          >
            Place Order
          </button>
        </div>
      </div>
    </div>
  );
};


OrderDetails.propTypes = {
  // eslint-disable-next-line
  orders: PropTypes.array,
  total: PropTypes.number.isRequired,
  displayModal: PropTypes.func.isRequired,
  handleRemove: PropTypes.func.isRequired,
};


export default OrderDetails;
