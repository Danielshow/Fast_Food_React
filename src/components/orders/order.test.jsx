import React from 'react';
import { shallow } from 'enzyme';
import '../../__test__/setup/setupEnzyme';
import Orders from './Order';
import {mapStateToProps, mapDispatchToProps} from './OrderContainer';
import OrderDetails from './OrderDetails';
import OrderConfirmation from './orderConfirmation';
import combineOrders from '../../helpers/orderHelpers';

describe('### Order Component', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<Orders />);
  });

  it('should display ordercontainer class', () => {
    const container = wrapper.find('.orderContainer');
    expect(container.length).toEqual(1);
  });

  it('should return spinner if loading is true', () => {
    wrapper.setProps({'loading': true});
    const container = wrapper.find('Spinner');
    expect(container.length).toEqual(1);
  });

  it('return H3 if foods props is empty', () => {
    wrapper.setProps({foods: []});
    const container = wrapper.find('h3');
    expect(container.length).toEqual(1);
  });

  it('should call orderConfirmation function when it loads', () => {
    wrapper.setState({modal: true});
    const container = wrapper.find('OrderConfirmation');
    expect(container.length).toEqual(1);
  });

  it('should call function when it loads', () => {
    wrapper.setProps({'getFoods': ()=>{}});
    expect(wrapper.props().getFoods).toBeCalled;
  });

  it('should return button if food is supplied', () => {
    wrapper.setProps({foods: [{id: 1, }, {id: 2}]});
    const container = wrapper.find('button');
    expect(container.length).toEqual(2);
  });

  it('should simulate button click if food is supplied', () => {
    const handleClick = jest.fn();
    wrapper.setProps({foods: [{id: 1, }, {id: 2}]});
    const container = wrapper.find('button');
    container.at(0).simulate('click');
    expect(handleClick).toBeCalled;
  });

  it('should food css class', () => {
    wrapper.setProps({
      foods: [
        {id: 1},
        {id: 2}
      ]
    });

    const container = wrapper.find('.food');
    expect(container.length).toEqual(2);
  });

  it('should find foodItems css class', () => {
    wrapper.setProps({
      foods: [
        {id: 1},
        {id: 2}
      ]
    });

    const container = wrapper.find('.foodImage');
    expect(container.length).toEqual(2);
  });

  it('should find foodItems css class', () => {
    wrapper.setProps({
      foods: [
        {id: 1},
        {id: 2}
      ],
      onClick: jest.fn()
    });

    const container = wrapper.find('button');
    container.at(0).simulate('click');
    expect(wrapper.props().onClick).toBeCalled;
  });
});

describe('##Order Container', () => {
  const state = {
    ord: {
      foods: [],
      loading: false
    }
  };

  it('should return mapStateToProps', () => {
    expect(typeof mapStateToProps(state)).toEqual('object');
  });

  it('should return mapDispatchToProps', () => {
    expect(typeof mapDispatchToProps(state)).toEqual('object');
  });
});

describe('### OrderDetails component', () => {
  let wrapper;
  const orders = [{id:1}, {id:2}];
  const total = 0;
  const handleRemove=() => {};
  const displayModal =() => {};
  beforeAll(() => {
    wrapper = shallow(<OrderDetails
      orders={orders}
      total={total}
      displayModal={displayModal}
      handleRemove={handleRemove}
    />);
  });

  it('should simulate click', () => {
    const button = wrapper.find('button');
    button.at(0).simulate('click');
    expect(wrapper.prop('displayModal')).toBeCalled;
  });

  it('should simulate span click', () => {
    const span = wrapper.find('span.remove');
    span.at(0).simulate('click');
    expect(wrapper.props().handleRemove).toBeCalled;
  });
});


describe('## Order comfirmation component', () => {
  let wrapper;
  const removeModal = ()=>{};
  const total = 40;
  const orderFood = jest.fn();
  beforeAll(() => {
    wrapper = shallow(
      <OrderConfirmation
        removeModal={removeModal}
        total={total}
        orderFood={orderFood}
      />);
  });

  it('should display Toast notification if some fields atre invalid', () =>{
    wrapper.setState({
      perror: true,
      addressError: true
    });

    const toast = wrapper.find('Toast');
    expect(toast.length).toEqual(1);
  });

  it('should render Header', () => {
    const header = wrapper.find('h3');
    expect(header.length).toEqual(2);
  });

  it('should render removeModal props when h1 is clicked', () => {
    const header = wrapper.find('h1');
    header.at(0).simulate('click');
    expect(wrapper.props('removeModal')).toBeCalled;
  });

  it('should change the state if invalid phonenumber field is supplied', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'phonenumber', value: '09888888888888'}});
    expect(wrapper.state().phonenumber).toEqual('09888888888888');
    expect(wrapper.state().perror).toEqual(true);
  });

  it('should change the state if phonenumber field is changed', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'phonenumber', value: '09888888888'}});
    expect(wrapper.state().phonenumber).toEqual('09888888888');
    expect(wrapper.state().perror).toEqual(false);
  });

  it('should change the state after address field is changed', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'address', value: 'hhhhh'}});
    expect(wrapper.state().address).toEqual('hhhhh');
    expect(wrapper.state().addressError).toEqual(false);
  });

  it('should change the state if address field is invalid', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'address', value: 'hhhh'}});
    expect(wrapper.state().address).toEqual('hhhh');
    expect(wrapper.state().addressError).toEqual(true);
  });

  it('should change the state after invalid address field', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'address', value: 'hhhh'}});
    expect(wrapper.state().addressError).toEqual(true);
  });

  it('should change the state after invalid id', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'addres', value: 'hhhh'}});
    expect(wrapper.state().addres).toEqual('hhhh');
  });

  it('should change the state after invalid phonenumber field', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'phonenumber', value: '900900'}});
    expect(wrapper.state().perror).toEqual(true);
  });

  it('should return error message if error response is not null', () => {
    wrapper.setState({
      response: 'yeah'
    });

    const paragraph = wrapper.find('p');
    expect(paragraph.length).toEqual(2);
  });


  it('should return error if submit button is clicked', () => {
    wrapper.setState({perror: true, addressError: true});
    const fakeEvent = { preventDefault: () => {}};
    const form = wrapper.find('form').simulate('submit', fakeEvent);
    expect(form).toBeCalled;
  });

  it('should submit if submit button is clicked', () => {
    wrapper.setState({address: 'ddj', phonenumber: 'dkdjdj'});
    const fakeEvent = { preventDefault: () => {}};
    wrapper.find('form').simulate('submit', fakeEvent);
    expect(combineOrders([],
      wrapper.state().address, wrapper.state().phonenumber)).toBeCalled;
    expect(wrapper.props().orderFood).toBeCalled;
  });
});

describe('### OrderDetails component', () => {
  let wrapper;
  const orders = [{id:1}, {id:2}];
  const total = 0;
  const handleRemove=() => {};
  const displayModal =() => {};
  beforeAll(() => {
    wrapper = shallow(<OrderDetails
      orders={orders}
      total={total}
      displayModal={displayModal}
      handleRemove={handleRemove}
    />);
  });

  it('should simulate click', () => {
    const button = wrapper.find('button');
    button.at(0).simulate('click');
    expect(wrapper.prop('displayModal')).toBeCalled;
  });

  it('should simulate span click', () => {
    const span = wrapper.find('span.remove');
    span.at(0).simulate('click');
    expect(wrapper.props().handleRemove).toBeCalled;
  });
});


describe('## Order comfirmation component', () => {
  let wrapper;
  const removeModal = ()=>{};
  const total = 40;
  const orderFood = jest.fn();
  beforeAll(() => {
    wrapper = shallow(
      <OrderConfirmation
        removeModal={removeModal}
        total={total}
        orderFood={orderFood}
      />);
  });

  it('should display Toast notification if some fields atre invalid', () =>{
    wrapper.setState({
      perror: true,
      addressError: true
    });

    const toast = wrapper.find('Toast');
    expect(toast.length).toEqual(1);
  });

  it('should render Header', () => {
    const header = wrapper.find('h3');
    expect(header.length).toEqual(2);
  });

  it('should render removeModal props when h1 is clicked', () => {
    const header = wrapper.find('h1');
    header.at(0).simulate('click');
    expect(wrapper.props('removeModal')).toBeCalled;
  });

  it('should change the state if invalid phonenumber field is supplied', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'phonenumber', value: '09888888888888'}});
    expect(wrapper.state().phonenumber).toEqual('09888888888888');
    expect(wrapper.state().perror).toEqual(true);
  });

  it('should change the state if phonenumber field is changed', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'phonenumber', value: '09888888888'}});
    expect(wrapper.state().phonenumber).toEqual('09888888888');
    expect(wrapper.state().perror).toEqual(false);
  });

  it('should change the state after address field is changed', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'address', value: 'hhhhh'}});
    expect(wrapper.state().address).toEqual('hhhhh');
    expect(wrapper.state().addressError).toEqual(false);
  });

  it('should change the state if address field is invalid', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'address', value: 'hhhh'}});
    expect(wrapper.state().address).toEqual('hhhh');
    expect(wrapper.state().addressError).toEqual(true);
  });

  it('should change the state after invalid address field', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'address', value: 'hhhh'}});
    expect(wrapper.state().addressError).toEqual(true);
  });

  it('should change the state after invalid id', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'addres', value: 'hhhh'}});
    expect(wrapper.state().addres).toEqual('hhhh');
  });

  it('should change the state after invalid phonenumber field', () => {
    const input = wrapper.find('input');
    input.at(0).simulate(
      'change', {target: {id: 'phonenumber', value: '900900'}});
    expect(wrapper.state().perror).toEqual(true);
  });

  it('should return error message if error response is not null', () => {
    wrapper.setState({
      response: 'yeah'
    });

    const paragraph = wrapper.find('p');
    expect(paragraph.length).toEqual(2);
  });


  it('should return error if submit button is clicked', () => {
    wrapper.setState({perror: true, addressError: true});
    const fakeEvent = { preventDefault: () => {}};
    const form = wrapper.find('form').simulate('submit', fakeEvent);
    expect(form).toBeCalled;
  });

  it('should submit if submit button is clicked', () => {
    wrapper.setState({address: 'ddj', phonenumber: 'dkdjdj'});
    const fakeEvent = { preventDefault: () => {}};
    wrapper.find('form').simulate('submit', fakeEvent);
    expect(combineOrders([],
      wrapper.state().address, wrapper.state().phonenumber)).toBeCalled;
    expect(wrapper.props().orderFood).toBeCalled;
  });
});
