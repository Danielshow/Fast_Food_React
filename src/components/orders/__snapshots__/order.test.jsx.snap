// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`## Order comfirmation component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderConfirmation
    orderFood={[MockFunction]}
    removeModal={[Function]}
    total={40}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="orderwidth"
      >
        <Toast
          warning="Some fields are invalid"
        />
        <h1
          className="close"
          onClick={[Function]}
        >
           x 
        </h1>
        <h3>
          Order Details
        </h3>
        <h3>
          Total:  
          40
        </h3>
        <form
          className="orderDetailsContainer"
          onSubmit={[Function]}
        >
          <div
            className="input-field"
          >
            <label
              htmlFor="address"
            >
              Address
            </label>
            <br />
            <input
              className="inValid"
              id="address"
              onChange={[Function]}
              required={true}
              type="text"
              value="ddj"
            />
          </div>
          <div
            className="input-field"
          >
            <label
              htmlFor="phonenumber"
            >
              Phone Number
            </label>
            <br />
            <input
              className="inValid"
              id="phonenumber"
              onChange={[Function]}
              required={true}
              type="text"
              value="dkdjdj"
            />
          </div>
          <p
            className="errorMessage"
          >
            yeah
          </p>
          <button
            type="submit"
          >
             Order 
          </button>
        </form>
        <p>
           Thanks for patronizing us
        </p>
      </div>,
      "className": "orderdetails",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Toast
            warning="Some fields are invalid"
          />,
          <h1
            className="close"
            onClick={[Function]}
          >
             x 
          </h1>,
          <h3>
            Order Details
          </h3>,
          <h3>
            Total:  
            40
          </h3>,
          <form
            className="orderDetailsContainer"
            onSubmit={[Function]}
          >
            <div
              className="input-field"
            >
              <label
                htmlFor="address"
              >
                Address
              </label>
              <br />
              <input
                className="inValid"
                id="address"
                onChange={[Function]}
                required={true}
                type="text"
                value="ddj"
              />
            </div>
            <div
              className="input-field"
            >
              <label
                htmlFor="phonenumber"
              >
                Phone Number
              </label>
              <br />
              <input
                className="inValid"
                id="phonenumber"
                onChange={[Function]}
                required={true}
                type="text"
                value="dkdjdj"
              />
            </div>
            <p
              className="errorMessage"
            >
              yeah
            </p>
            <button
              type="submit"
            >
               Order 
            </button>
          </form>,
          <p>
             Thanks for patronizing us
          </p>,
        ],
        "className": "orderwidth",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "warning": "Some fields are invalid",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " x ",
            "className": "close",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " x ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Order Details",
          },
          "ref": null,
          "rendered": "Order Details",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Total:  ",
              40,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Total:  ",
            40,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="input-field"
              >
                <label
                  htmlFor="address"
                >
                  Address
                </label>
                <br />
                <input
                  className="inValid"
                  id="address"
                  onChange={[Function]}
                  required={true}
                  type="text"
                  value="ddj"
                />
              </div>,
              <div
                className="input-field"
              >
                <label
                  htmlFor="phonenumber"
                >
                  Phone Number
                </label>
                <br />
                <input
                  className="inValid"
                  id="phonenumber"
                  onChange={[Function]}
                  required={true}
                  type="text"
                  value="dkdjdj"
                />
              </div>,
              <p
                className="errorMessage"
              >
                yeah
              </p>,
              <button
                type="submit"
              >
                 Order 
              </button>,
            ],
            "className": "orderDetailsContainer",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="address"
                  >
                    Address
                  </label>,
                  <br />,
                  <input
                    className="inValid"
                    id="address"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="ddj"
                  />,
                ],
                "className": "input-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Address",
                    "htmlFor": "address",
                  },
                  "ref": null,
                  "rendered": "Address",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "inValid",
                    "id": "address",
                    "onChange": [Function],
                    "required": true,
                    "type": "text",
                    "value": "ddj",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="phonenumber"
                  >
                    Phone Number
                  </label>,
                  <br />,
                  <input
                    className="inValid"
                    id="phonenumber"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="dkdjdj"
                  />,
                ],
                "className": "input-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Phone Number",
                    "htmlFor": "phonenumber",
                  },
                  "ref": null,
                  "rendered": "Phone Number",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "inValid",
                    "id": "phonenumber",
                    "onChange": [Function],
                    "required": true,
                    "type": "text",
                    "value": "dkdjdj",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "yeah",
                "className": "errorMessage",
              },
              "ref": null,
              "rendered": "yeah",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Order ",
                "type": "submit",
              },
              "ref": null,
              "rendered": " Order ",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Thanks for patronizing us",
          },
          "ref": null,
          "rendered": " Thanks for patronizing us",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="orderwidth"
        >
          <Toast
            warning="Some fields are invalid"
          />
          <h1
            className="close"
            onClick={[Function]}
          >
             x 
          </h1>
          <h3>
            Order Details
          </h3>
          <h3>
            Total:  
            40
          </h3>
          <form
            className="orderDetailsContainer"
            onSubmit={[Function]}
          >
            <div
              className="input-field"
            >
              <label
                htmlFor="address"
              >
                Address
              </label>
              <br />
              <input
                className="inValid"
                id="address"
                onChange={[Function]}
                required={true}
                type="text"
                value="ddj"
              />
            </div>
            <div
              className="input-field"
            >
              <label
                htmlFor="phonenumber"
              >
                Phone Number
              </label>
              <br />
              <input
                className="inValid"
                id="phonenumber"
                onChange={[Function]}
                required={true}
                type="text"
                value="dkdjdj"
              />
            </div>
            <p
              className="errorMessage"
            >
              yeah
            </p>
            <button
              type="submit"
            >
               Order 
            </button>
          </form>
          <p>
             Thanks for patronizing us
          </p>
        </div>,
        "className": "orderdetails",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Toast
              warning="Some fields are invalid"
            />,
            <h1
              className="close"
              onClick={[Function]}
            >
               x 
            </h1>,
            <h3>
              Order Details
            </h3>,
            <h3>
              Total:  
              40
            </h3>,
            <form
              className="orderDetailsContainer"
              onSubmit={[Function]}
            >
              <div
                className="input-field"
              >
                <label
                  htmlFor="address"
                >
                  Address
                </label>
                <br />
                <input
                  className="inValid"
                  id="address"
                  onChange={[Function]}
                  required={true}
                  type="text"
                  value="ddj"
                />
              </div>
              <div
                className="input-field"
              >
                <label
                  htmlFor="phonenumber"
                >
                  Phone Number
                </label>
                <br />
                <input
                  className="inValid"
                  id="phonenumber"
                  onChange={[Function]}
                  required={true}
                  type="text"
                  value="dkdjdj"
                />
              </div>
              <p
                className="errorMessage"
              >
                yeah
              </p>
              <button
                type="submit"
              >
                 Order 
              </button>
            </form>,
            <p>
               Thanks for patronizing us
            </p>,
          ],
          "className": "orderwidth",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "warning": "Some fields are invalid",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " x ",
              "className": "close",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": " x ",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Order Details",
            },
            "ref": null,
            "rendered": "Order Details",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Total:  ",
                40,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Total:  ",
              40,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="input-field"
                >
                  <label
                    htmlFor="address"
                  >
                    Address
                  </label>
                  <br />
                  <input
                    className="inValid"
                    id="address"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="ddj"
                  />
                </div>,
                <div
                  className="input-field"
                >
                  <label
                    htmlFor="phonenumber"
                  >
                    Phone Number
                  </label>
                  <br />
                  <input
                    className="inValid"
                    id="phonenumber"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="dkdjdj"
                  />
                </div>,
                <p
                  className="errorMessage"
                >
                  yeah
                </p>,
                <button
                  type="submit"
                >
                   Order 
                </button>,
              ],
              "className": "orderDetailsContainer",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="address"
                    >
                      Address
                    </label>,
                    <br />,
                    <input
                      className="inValid"
                      id="address"
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value="ddj"
                    />,
                  ],
                  "className": "input-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Address",
                      "htmlFor": "address",
                    },
                    "ref": null,
                    "rendered": "Address",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "inValid",
                      "id": "address",
                      "onChange": [Function],
                      "required": true,
                      "type": "text",
                      "value": "ddj",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="phonenumber"
                    >
                      Phone Number
                    </label>,
                    <br />,
                    <input
                      className="inValid"
                      id="phonenumber"
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value="dkdjdj"
                    />,
                  ],
                  "className": "input-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Phone Number",
                      "htmlFor": "phonenumber",
                    },
                    "ref": null,
                    "rendered": "Phone Number",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "inValid",
                      "id": "phonenumber",
                      "onChange": [Function],
                      "required": true,
                      "type": "text",
                      "value": "dkdjdj",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "yeah",
                  "className": "errorMessage",
                },
                "ref": null,
                "rendered": "yeah",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Order ",
                  "type": "submit",
                },
                "ref": null,
                "rendered": " Order ",
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Thanks for patronizing us",
            },
            "ref": null,
            "rendered": " Thanks for patronizing us",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
